client/state/index.js

```diff
@@ -109,7 +109,8 @@ const middleware = [ thunkMiddleware, noticesMiddleware ];
 if ( typeof window === 'object' ) {
 	// Browser-specific middlewares
 	middleware.push(
-		require( './analytics/middleware.js' ).analyticsMiddleware
+		require( './analytics/middleware.js' ).analyticsMiddleware,
+		require( './data-layer/wpcom-api-middleware.js' ).middleware,
```

Comment:
>  could happen synchronously even.

You don't want to do anything sync in a node handler. It blocks the server from handling other requests while the request is pending.